Add compatibility with std::pmr

ktf opened the pull request:
The move assign operator is deleted is std::pmr::polymorphic_allocator so we cannot iterate
anymore at runtime on the headers in the stack and move them.

alibuild commented:
Error while checking build/O2/fullCI_slc9 for 855ea9c14b2a72baa8dc76676c32700e0ae8e630 at 2025-06-10 15:53:
```
## sw/BUILD/O2-latest/log
/sw/SOURCES/O2/14379-slc9_x86-64/0/Detectors/TPC/workflow/src/ClusterDecoderRawSpec.cxx:89:20: error: no matching function for call to 'swap(o2::header::Stack&, o2::header::Stack&)'
/sw/slc9_x86-64/GCC-Toolchain/v14.2.0-alice2-1/include/c++/14.2.0/type_traits:138:11: error: no type named 'type' in 'struct std::enable_if<false, void>'
/sw/SOURCES/O2/14379-slc9_x86-64/0/Detectors/TPC/workflow/src/ClusterDecoderRawSpec.cxx:98:18: error: no matching function for call to 'swap(o2::header::Stack&, o2::header::Stack&)'
ninja: build stopped: subcommand failed.
```
Full log [here](https://ali-ci.cern.ch/alice-build-logs/AliceO2Group/AliceO2/14379/855ea9c14b2a72baa8dc76676c32700e0ae8e630/build_O2_fullCI_slc9/pretty.html).

ktf committed:
Add compatibility with std::pmr

The move assign operator is deleted is std::pmr::polymorphic_allocator so we cannot iterate
anymore at runtime on the headers in the stack and move them.

ktf commented:
@knopers8 @Barthelemy any objections to this? I assume we will never have more than 8 headers to copy...

knopers8 commented on Utilities/DataSampling/src/Dispatcher.cxx:
```diff
@@ -77,6 +78,42 @@ void Dispatcher::init(InitContext& ctx)
 
+header::Stack extractAdditionalHeaders(const char* inputHeaderStack)
+{
+  std::array<header::BaseHeader const*, 5> headers;
```
@ktf here, you used `5`, but the code below supports up to `8`. A typo?

ktf replied:
Yes. It used to be 5 and I then changed it to 8. Will fix this.
